### ISSUE TITLE
variable scaling for multiple nonlinear systems

### ISSUE THREAD

**Issue by lynnmunday** [idaholab/moose#25646]
## Reason
I don't think scaling works for multiple nonlinear systems.  I get this error:

```
libMesh terminating:
map_find() error: key "scaling_factors" not found in file ../src/systems/system.C on line 897
[0] ./include/libmesh/utility.h, line 136, compiled Sep 26 2023 at 16:52:00
```
I would like to be able to apply scaling for individual variables.   automatic_scaling in the executioner block would also be nice but I can probably achieve the same affect with individual scaling.  

## Design
I'll attach an input file showing the error.

## Impact
none

**Comment by lynnmunday** [comment]
```

[Problem]
  nl_sys_names = 'nl0 adjoint'
  kernel_coverage_check = false
[]

[Mesh]
  [gmg]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 4
    ny = 2
    nz = 1
    xmin = 9.4615
    xmax = 92.0115
    ymin = 3.175
    ymax = 22.225
    zmin = 0.489
    zmax = 0.755
  []
[]

[Variables]
  [T]
    scaling=1e3
  []
  [lam_T]
    nl_sys = adjoint
  []
[]

[Kernels]
  [heat_conduction]
    type = ADMatDiffusion
    variable = T
    diffusivity = thermal_conductivity
  []
  [heat_source]
    type = ADBodyForce
    function = src_fuel_function
    variable = T
  []
[]

[BCs]
  [conv_BC_front]
    type = ADConvectiveHeatFluxBC
    variable = T
    boundary = front
    T_infinity = 355
    heat_transfer_coefficient = 90000
  []
  [conv_BC_back]
    type = ADConvectiveHeatFluxBC
    variable = T
    boundary = back
    T_infinity = 325
    heat_transfer_coefficient = 40000
  []
[]

[Materials]
  # fuel properties
  [fuel_thermal]
    type = ADGenericConstantMaterial
    prop_names = thermal_conductivity
    prop_values = 17.6e3
  []
[]

[Executioner]
  type = SteadyAndAdjoint
  forward_system = nl0
  adjoint_system = adjoint

  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
  petsc_options_value = 'lu mumps'

  # automatic_scaling = true
  # scaling_group_variables = 'u_x u_y u_z'
  # off_diagonals_in_auto_scaling = true
  # compute_scaling_once = false

  line_search = 'none'

  nl_rel_tol = 1e-6
  nl_abs_tol = 1e-30
  nl_max_its = 10

  # l_tol = 1e-4
  l_max_its = 10
[]

##---------Forward Optimization stuff------------------#
[Reporters]
  [measure_data]
    type = ConstantReporter
    real_vector_names = 'x y z u weight'
    real_vector_values = '0.2 0.2 0.0; 0.3 0.8 0.0; 0 0 0; 5 5 5; 1 1 1'
  []
  [params_fuel]
    type = ConstantReporter
    real_vector_names = 'source'
    real_vector_values = '5e7' # Dummy
  []
[]

[Functions]
  [src_fuel_function]
    type = ParsedOptimizationFunction
    expression = q
    param_symbol_names = 'q'
    param_vector_name = 'params_fuel/source'
  []
[]
##---------Adjoint Optimization stuff------------------#
[DiracKernels]
  [adjointLoad_T]
    type = ReporterPointSource
    variable = lam_T
    x_coord_name = measure_data/x
    y_coord_name = measure_data/y
    z_coord_name = measure_data/z
    value_name = measure_data/u
    # weight_name = misfit/weighted_temperature
  []
[]
##--------- Outputs ------------------#

[Debug]
  show_var_residual_norms = true
[]
```

**Comment by lindsayad** [comment]
thanks for the report @lynnmunday !

**Comment by MengnanLi91** [comment]
I can look into this @lindsayad

**Comment by lindsayad** [comment]
great!

**Comment by GiudGiud** [comment]
Fixed by #26282
